dlc-oracle/src/main/scala/org/bitcoins/dlc/oracle/DLCOracleAppConfig.scala

```diff
@@ -61,8 +65,41 @@ case class DLCOracleAppConfig(
   }
 
   def exists(): Boolean = {
-    seedExists() &&
-    Files.exists(baseDatadir.resolve("oracle.sqlite"))
+    lazy val hasDb = this.driver match {
+      case PostgreSQL => true
+      case SQLite =>
+        Files.exists(baseDatadir.resolve("oracle.sqlite"))
+    }
+    seedExists() && hasDb
+  }
+
+  def initialize(oracle: DLCOracle): Future[DLCOracle] = {
+    val result =
+      FutureUtil.foldLeftAsync((), allTables)((_, table) => createTable(table))
+
+    result.failed.foreach(_.printStackTrace())
+
+    result.map(_ => oracle)
+  }
+
+  def initialize(
+      password: AesPassword,
+      bip39PasswordOpt: Option[String] = None): Future[DLCOracle] = {
+    if (!seedExists()) {
+      val entropy = MnemonicCode.getEntropy256Bits
+      val mnemonicCode = MnemonicCode.fromEntropy(entropy)
+      val decryptedMnemonic = DecryptedMnemonic(mnemonicCode, TimeUtil.now)
+      val encrypted = decryptedMnemonic.encrypt(password)
+      WalletStorage.writeMnemonicToDisk(seedPath, encrypted)
+    }
+
+    val key =
+      WalletStorage.getPrivateKeyFromDisk(seedPath,
+                                          SegWitMainNetPriv,
```

Comment:
is this param not used or else is there any way to ensure that mainnet is really the intended target?

Reply:
We just needed to have a private key version, using this one because the address we will staking under will be mainnet native segwit